src/services/content.ts

```diff
@@ -0,0 +1,344 @@
+ */
+const dispatchFeedItem = (
+  dispatch: Dispatch,
+  message: BroadcastAnnouncement | ReplyAnnouncement,
+  content: ActivityContentNote,
+  blockNumber: number
+) => {
+  const decoder = new TextDecoder();
+  dispatch(
+    addFeedItem({
+      fromId: decoder.decode((message.fromId as any) as Uint8Array),
+      blockNumber: blockNumber,
+      hash: decoder.decode((message.contentHash as any) as Uint8Array),
+      published: content.published,
+      uri: decoder.decode((message.url as any) as Uint8Array),
+      content: content,
+      inReplyTo:
+        message.announcementType === core.announcements.AnnouncementType.Reply
+          ? decoder.decode((message.inReplyTo as any) as Uint8Array)
+          : undefined,
+    })
+  );
+};
+
+/**
+ * dispatchProfile dispatches a profile to the redux store
+ * @param dispatch function used to dispatch to store
+ * @param message announcment information from batch
+ * @param profile profile to dispatch
+ * @param blockNumber number of block containing the publication
+ * @param blockIndex index of log message (relative to other DSNP logs) within the block
+ * @param batchIndex index of announcment within the batch
+ */
+const dispatchProfile = (
+  dispatch: Dispatch,
+  message: BroadcastAnnouncement,
+  profile: ActivityContentProfile,
+  blockNumber: number,
+  blockIndex: number,
+  batchIndex: number
+) => {
+  const decoder = new TextDecoder();
+
+  dispatch(
+    upsertProfile({
+      ...profile,
+      fromId: decoder.decode((message.fromId as any) as Uint8Array),
+      blockNumber,
+      blockIndex,
+      batchIndex,
+    } as Profile)
+  );
+};
+
+/**
+ * handleRegistryUpdate dispatches a profile update for the handle when a registry update is made.
+ * @param dispatch function used to dispatch to store
+ * @param update registry log message containg DSNP uri and handle
+ */
+const handleRegistryUpdate = (dispatch: Dispatch) => (
+  update: RegistryUpdateLogData
+) => {
+  dispatch(
+    upsertProfile({
+      ...createProfile(),
+      fromId: core.identifiers.convertDSNPUserURIToDSNPUserId(
+        update.dsnpUserURI
+      ),
+      handle: update.handle,
+    } as Profile)
+  );
+};
+
+/**
+ * fetchAndDispatchContent retrieves activity content for an annoucement and then dispatches
+ * it to the redux store.
+ * @param dispatch function used to dispatch to store
+ * @param message broadcast annoucnement containing activity content url
+ * @param blockNumber block number of block containing publication
+ * @param blockIndex index of publication within block
+ * @param batchIndex index of announcment within batch
+ * @throws if content cannot be retrieved
+ */
+const fetchAndDispatchContent = async (
+  dispatch: Dispatch,
+  message: BroadcastAnnouncement,
+  blockNumber: number,
+  blockIndex: number,
+  batchIndex: number
+) => {
+  const decoder = new TextDecoder();
+
+  try {
+    const url = decoder.decode((message.url as any) as Uint8Array);
```

Comment:
Conflict and rebase warning with #81 which ever gets in first

Reply:
#81 has gone away. #86 supersedes it and is already rebased on this PR.